demos/imagebuilder-windowsbaseline/main.bicep

```diff
@@ -0,0 +1,212 @@
+param _artifactsLocation string = deployment().properties.templateLink.uri
+
+@description('The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated. Use the defaultValue if the staging location is not secured.')
+@secure()
+param _artifactsLocationSasToken string = ''
+
+@description('The Azure region where resources in the template should be deployed.')
+param location string = resourceGroup().location
+
+@description('The name of the customizer script which will be executed during image build.')
+param customizerScriptName string = 'scripts/runScript.ps1'
+
+@description('Name of the user-assigned managed identity used by Azure Image Builder template, and for triggering the Azure Image Builder build at the end of the deployment')
+param identityName string = substring('ImageGallery_${guid(resourceGroup().id)}', 0, 21)
+
+@description('Permissions to allow for the user-assigned managed identity.')
+param roleDefinitionName string = guid(resourceGroup().id)
+
+@description('Name of the role to grant for the user-assigned managed identity.')
+param roleAssignmentName string = guid(resourceGroup().id)
+
+@description('Name of the new Azure Image Gallery resource.')
+param imageGalleryName string = substring('ImageGallery_${guid(resourceGroup().id)}', 0, 21)
+
+@description('Detailed image information to set for the custom image produced by the Azure Image Builder build.')
+param imageDefinition object = {
+  imageDefinitionName: 'Win2019_AzureWindowsBaseline_Definition'
+  publisher: 'AzureWindowsBaseline'
+  offer: 'WindowsServer'
+  sku: '2019-Datacenter'
+}
+
+@description('Name of the template to create in Azure Image Builder.')
+param imageTemplateName string = 'Win2019_AzureWindowsBaseline_Template'
+
+@description('Name of the custom iamge to create and distribute using Azure Image Builder.')
+param runOutputName string = 'Win2019_AzureWindowsBaseline_CustomImage'
+
+@description('List the regions in Azure where you would like to replicate the custom image after it is created.')
+param replicationRegions array = [
+  'centralus'
+  'eastus'
+  'eastus2'
+  'westus'
+  'westus2'
+  'westus3'
+  'northcentralus'
+  'southcentralus'
+  'westcentralus'
+]
+
+var customizerScriptUri = uri(_artifactsLocation, '${customizerScriptName}${_artifactsLocationSasToken}')
+
+resource identityName_resource 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: identityName
+  location: location
+}
+
+resource roleDefinitionName_resource 'Microsoft.Authorization/roleDefinitions@2018-07-01' = {
+  name: roleDefinitionName
+  properties: {
+    roleName: roleDefinitionName
+    description: 'Used for AIB template and ARM deployment script that runs AIB build'
+    type: 'customRole'
+    isCustom: true
+    permissions: [
+      {
+        actions: [
```

Comment:
are we creating a built-in definition for this?  (eventually?)

Reply:
investigating